kiali/kiali
# Add separate cluster for configured kiali instance

** Describe the change **

This change lets you manually specify a cluster in the kiali config that there is no remote secret for. This cluster is considered "inaccessible" meaning that Kiali won't try to connect to it and the frontend won't consider it for the "multi-cluster mode" e.g. there won't be a cluster column when there's one accessible cluster and one inaccessible cluster.

To test:
1. Create two clusters. You can use: `hack/run-integration-tests.sh --test-suite frontend-multi-primary --setup-only true`.
2. Deploy kiali on both clusters including the "west" cluster. `./hack/istio/multicluster/deploy-kiali.sh --single-kiali false --cluster1-context kind-east --cluster2-context kind-west --kiali-auth-strategy anonymous --manage-kind true --kiali-use-dev-image true`
3. Add the "east" cluster to the "west" kiali config by editing the "west" kiali configmap (`kubectl --context kind-west edit cm kiali -n istio-system`):
    ```
    clustering:
      clusters:
        - name: east
          accessible: false
      kiali_urls:
        - cluster_name: east
          url: http://<kiali_url>/kiali
          instance_name: kiali
          namespace: istio-system
    ```
4. Restart the west kiali: `kubectl --context kind-west delete pod -n istio-system -l app=kiali`
5. Navigate to the west kiali's "mesh" page (you can port-forward to it via `hack/kiali-port-forward.sh -kc kind-west`).
  You should see two kiali instances on the mesh page and a link to the "east" kiali:
![Screenshot from 2023-12-20 13-31-19](https://github.com/kiali/kiali/assets/6226732/e3dc7efe-667e-4700-8ace-a056e69100e0)
  You should _not_ see a cluster column on any of the app/service/istio/workload pages for the "west" kiali.

** Issue reference **

Fixes #6243
